File: java/parser/src/main/java/gov/lbl/parser/parser/Element_replaceableVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Component_clause1;
import gov.lbl.parser.domain.Constraining_clause;
import gov.lbl.parser.domain.Element_replaceable;
import gov.lbl.parser.domain.Short_class_definition;

public class Element_replaceableVisitor extends modelicaBaseVisitor<Element_replaceable> {
    @Override
    public Element_replaceable visitElement_replaceable(modelicaParser.Element_replaceableContext ctx) {
        Short_class_definitionVisitor short_class_definitionVisitor = new Short_class_definitionVisitor();
        Short_class_definition short_class_definition = ctx.short_class_definition() == null ? null : ctx.short_class_definition().accept(short_class_definitionVisitor);
        
        Component_clause1Visitor component_clause1Visitor = new Component_clause1Visitor();
        Component_clause1 component_clause1 = ctx.component_clause1() == null ? null : ctx.component_clause1().accept(component_clause1Visitor);

        Constraining_clauseVisitor constraining_clauseVisitor = new Constraining_clauseVisitor();
        Constraining_clause constraining_clause = ctx.constraining_clause() == null ? null : ctx.constraining_clause().accept(constraining_clauseVisitor);
        
        return new Element_replaceable(short_class_definition, component_clause1, constraining_clause);
    }
}

File: jsParser/parser/Element_replaceableVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor;
const Element_replaceable = require('../domain/Element_replaceable');

const Short_class_definitionVisitor = require('./Short_class_definitionVisitor');
const Component_clause1Visitor = require('./Component_clause1Visitor');
const Constraining_clauseVisitor = require('./Constraining_clauseVisitor');

class Element_replaceableVisitor {
    constructor() {
        modelicaVisitor.call(this);
        return this;
    }
    visitElement_replaceable(ctx) {
        var short_class_definition = null;
        var component_clause1 = null;
        var constraining_clause = null;

        if (ctx.short_class_definition()) {
            var short_class_definitionVisitor = new Short_class_definitionVisitor.Short_class_definitionVisitor();
            short_class_definition = short_class_definitionVisitor.visitShort_class_definition(ctx.short_class_definition());
        }
        if (ctx.component_clause1()) {
            var component_clause1Visitor = new Component_clause1Visitor.Component_clause1Visitor();
            component_clause1 = component_clause1Visitor.visitComponent_clause1(ctx.component_clause1());
        }
        if (ctx.constraining_clause()) {
            var constraining_clauseVisitor = new Constraining_clauseVisitor.Constraining_clauseVisitor();
            constraining_clause = constraining_clauseVisitor.visitConstraining_clause(ctx.constraining_clause());
        }
        
        return new Element_replaceable.Element_replaceable(short_class_definition, component_clause1, constraining_clause);
    }
};

Element_replaceableVisitor.prototype = Object.create(modelicaVisitor.prototype);

exports.visitElement_replaceable = this.visitElement_replaceable;
exports.Element_replaceableVisitor = Element_replaceableVisitor;